internal/ghinstance/host.go

```diff
@@ -55,19 +55,35 @@ func GraphQLEndpoint(hostname string) string {
 	if IsEnterprise(hostname) {
 		return fmt.Sprintf("https://%s/api/graphql", hostname)
 	}
+	if strings.EqualFold(hostname, localhost) {
+		return fmt.Sprintf("http://api.%s/graphql", hostname)
+	}
 	return fmt.Sprintf("https://api.%s/graphql", hostname)
 }
 
 func RESTPrefix(hostname string) string {
 	if IsEnterprise(hostname) {
 		return fmt.Sprintf("https://%s/api/v3/", hostname)
 	}
+	if strings.EqualFold(hostname, localhost) {
+		return fmt.Sprintf("http://api.%s/", hostname)
+	}
 	return fmt.Sprintf("https://api.%s/", hostname)
 }
 
 func GistPrefix(hostname string) string {
 	if IsEnterprise(hostname) {
 		return fmt.Sprintf("https://%s/gist/", hostname)
 	}
+	if strings.EqualFold(hostname, localhost) {
+		return fmt.Sprintf("http://%s/gist/", hostname)
```

Comment:
On my local dev instance, `http://github.localhost/gist` redirects to `http://gist.github.localhost`

```suggestion
		return fmt.Sprintf("http://gist.%s/", hostname)
```

Reply:
@joshmgross Weird, for me (under Codespaces) the canonical location seems to be `github.localhost/gist`. The subdomain works fine, as well. But would you say that it's fine to generate links in the format of `http://github.localhost/gist/ID` in either environment and trust that the server will redirect them properly based on environment settings? I think the priority is just to not generate a broken link.

Reply:
👍  Sounds good to me